File: JavaScript_Courses/JavaScript_Basics/Preparation/06_Nested_Loops_Lab/04_Sum_of_Two_Numbers.js
function sumOfTwoNumbers(input) {
    let intervalStart = Number(input[0]);
    let intervalEnd = Number(input[1]);
    let magicNumber = Number(input[2]);

    let currentCombination = 0;
    let isFound = false;

    for (let n1 = intervalStart; n1 <= intervalEnd; n1++) {
        for (let n2 = intervalStart; n2 <= intervalEnd; n2++) {
            currentCombination++;

            currentSum = n1 + n2;

            if (currentSum == magicNumber) {
                console.log(`Combination N:${currentCombination} (${n1} + ${n2} = ${magicNumber})`);
                isFound = true;
                break;
            }
        }

        if (isFound) {
            break;
        }
    }

    // if (isFound === false) {
    if (!isFound) {
        console.log(`${currentCombination} combinations - neither equals ${magicNumber}`);
    }
}


// sumOfTwoNumbers(["1", "10", "5"]);

File: Python_Courses/Python_Basics/Python_PB_2024_06/06_Nested_Loops_Lab/04_Sum_of_Two_Numbers.py
start = int(input())
end = int(input())
magic_number = int(input())

comb_number = 0
n1 = 1
n2 = 1

should_break = False
for n1 in range(start, end + 1):
    for n2 in range(start, end + 1):
        comb_number += 1

        if n1 + n2 == magic_number:
            should_break = True
            break

    if should_break:
        break

if should_break:
    print(f'Combination N:{comb_number} ({n1} + {n2} = {magic_number})')
else:
    print(f'{comb_number} combinations - neither equals {magic_number}')
